azure_privatelink.go

```diff
@@ -0,0 +1,88 @@
+package aiven
+
+type (
+	// AzurePrivatelinkHandler is the client that interacts with the Azure Privatelink API on Aiven.
+	AzurePrivatelinkHandler struct {
+		client *Client
+	}
+
+	// AzurePrivatelinkRequest holds the parameters to create a new
+	// or update an existing Azure Privatelink.
+	AzurePrivatelinkRequest struct {
+		UserSubscriptionIDs []string `json:"user_subscription_ids"`
+	}
+
+	// AzurePrivatelinkResponse represents the response from Aiven after
+	// interacting with the Azure Privatelink.
+	AzurePrivatelinkResponse struct {
+		APIResponse
+		AzureServiceAlias   string   `json:"azure_service_alias"`
+		AzureServiceID      string   `json:"azure_service_id"`
+		Message             string   `json:"message"`
+		State               string   `json:"state"`
+		UserSubscriptionIDs []string `json:"user_subscription_ids"`
+	}
+)
+
+// Create creates an Azure Privatelink
+func (h *AzurePrivatelinkHandler) Create(project, serviceName string, subscriptionIDs []string) (*AzurePrivatelinkResponse, error) {
+	path := buildPath("project", project, "service", serviceName, "privatelink", "azure")
+	bts, err := h.client.doPostRequest(path, AzurePrivatelinkRequest{
+		UserSubscriptionIDs: subscriptionIDs,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	var rsp AzurePrivatelinkResponse
+	if err := checkAPIResponse(bts, &rsp); err != nil {
+		return nil, err
+	}
+
+	return &rsp, nil
+}
+
+// Update updates an Azure Privatelink
+func (h *AzurePrivatelinkHandler) Update(project, serviceName string, subscriptionIDs []string) (*AzurePrivatelinkResponse, error) {
```

Comment:
If we use a AzurePrivateLinkrequest struct as a parameter its easier to add fields later without breaking the API, so this is something we could consider, what do you think?